### PR TITLE
Add docs for usage of card reader flows in wcandroid

### PR DESCRIPTION
**Pull request by malinajirka**
### Description

This PR adds basic documentation for usage of existing In-Person Payment UI flows and helper classes.

### Testing instructions

No testing necessary
### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: docs/card-reader-module.md [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+## Connection Flow
+`CardReaderConnectDialogFragment` encapsulates the connection flow. Displays each step of the flow in a dialog. Returns a boolean flag (KEY_CONNECT_TO_READER_RESULT = "key_connect_to_reader_result") indicating whether the connection was successfully established to the original destination the flow was started from.
+
+## Payment Flow
+`CardReaderPaymentDialogFragment` encapsulates the payment flow. Displays each step of the flow in a dialog. A card reader needs to be already connected before this flow is started, otherwise finishes immediately and displays a Snackbar with an error message.
+
+`CardReaderPaymentCollectibilityChecker` should be used to verify whether an Order meets the requirements for IPP. If the app navigates to `CardReaderPaymentDialogFragment` for an order which is not eligible for IPP, the payment might fail with a generic error.
+
+# How to integrate CardReaderModule in another app
+
```
Comment (kidinov) [inline docs/card-reader-module.md:13]: This section is for the latter?

**Review by kidinov — APPROVED** [review]
Looks clear to me 👍. Left a question - please go ahead and merge it if the plan to feel the `How to integrate CardReaderModule in another app` section latter

File: docs/card-reader-module.md (hunk 1 above)
Comment (malinajirka) [reply docs/card-reader-module.md:13]: Nope, I meant it as a title for all the sub-titles below it -> it goes into details how to use the interfaces provided by the CardReaderModule. Any suggestions for a better naming are welcome :).

Comment (kidinov) [reply docs/card-reader-module.md:13]: Hm, then I'm confused:

image

In my understanding, the section about how to get started is about how to get started with working on the module e.g. for a newcomer

Comment (malinajirka) [reply docs/card-reader-module.md:13]: I see, you are right - I've updated the doc in https://github.com/woocommerce/woocommerce-android/pull/5158/commits/9f861c4b3d65639229dce0321d5716d461d1b94a. Wdyt is it better now? Feel free to commit changes into this PR if you have suggestion on how to reorganize it. Thanks!

**Comment by kidinov** [comment]
@malinajirka I will commit shortly my latest proposal. Feel free to revert if it doesn't look good

**Comment by kidinov** [comment]
@malinajirka I've done small reorderings of the section in here https://github.com/woocommerce/woocommerce-android/pull/5158/commits/84c0e1c73534b08ad8060231e1e30b2649d0f914 Please take a look

**Comment by malinajirka** [comment]
Hmm, I'm wondering if it'd make more sense to split it into two files. The order now is a bit misleading, since if a developer wants to re-use the existing UI flows, they don't need to understand any of the things mentioned above that section (eg. links in the GetStarted or the interfaces). Wdyt @kidinov ?

**Comment by kidinov** [comment]
@malinajirka I would keep it in 1 file for now just because it's pretty small. When we will add more info in there we could then build some trie like structure, e.g. the list of content which leads to a file with content

**Pull request merged**